outputs.tf

```diff
@@ -5,7 +5,7 @@ output "cluster_endpoint" {
 
 // Comma separated list of all DB instance endpoints running in cluster
 output "all_instance_endpoints_list" {
-  value = ["${aws_rds_cluster_instance.cluster_instance_0.endpoint}", "${aws_rds_cluster_instance.cluster_instance_n.*.endpoint}"]
+  value = ["${aws_rds_cluster_instance.cluster_instance_0.*.endpoint}", "${aws_rds_cluster_instance.cluster_instance_n.*.endpoint}"]
```

Comment:
This creates a list with 2 lists in it, unless Terraform does some weird auto flattening of lists (which it may well do, I don't know).

Use [concat](https://www.terraform.io/docs/configuration-0-11/interpolation.html#concat-list1-list2-) to combine lists into a single list:
```
value = ["${concat(aws_rds_cluster_instance.cluster_instance_0.*.endpoint, aws_rds_cluster_instance.cluster_instance_n.*.endpoint)}"]
```